command/monitor/monitor_test.go

```diff
@@ -0,0 +1,81 @@
+package monitor
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMonitor_Start(t *testing.T) {
```

Comment:
Feels like we should have a test that calls NewMonitor with buf==0, i.e. unbuffered.  Or prohibit that in NewMonitor.

Reply:
Will do

Reply:
I did some testing with this in the terminal, with `buf` set to `0` and there was a 25% drop rate for log messages, which felt like a terrible UX.  So I prohibited a `buf` of `0` in `NewMonitor` and added a corresponding test.

Reply:
> If the capacity is zero or absent, the channel is unbuffered and communication succeeds only when both a sender and receiver are ready. Otherwise, the channel is buffered and communication succeeds without blocking if the buffer is not full (sends) or not empty (receives)

Probably due to the above behavior. We don't wait to send on `Write`, so if the receiving end has not processed what's in the channel yet, the message gets dropped. 👍 on disallowing 0 as the buffer size.